webviz_subsurface/plugins/_co2_leakage/_utilities/initialization.py

```diff
@@ -121,25 +163,16 @@ def _find_max_file_size_mb(ens_path: str, table_rel_path: str) -> float:
 
 def init_menu_options(
     ensemble_roots: Dict[str, str],
-    mass_table: Dict[str, EnsembleTableProvider],
-    actual_volume_table: Dict[str, EnsembleTableProvider],
-    mass_relpath: str,
-    volume_relpath: str,
+    mass_table: Dict[str, ContainmentDataProvider],
+    actual_volume_table: Dict[str, ContainmentDataProvider],
+    unsmry_providers: Dict[str, UnsmryDataProvider],
 ) -> Dict[str, Dict[str, Dict[str, List[str]]]]:
-    options: Dict[str, Dict[str, Dict[str, List[str]]]] = {}
+    options: Dict[str, Dict[str, MenuOptions]] = {}
     for ens in ensemble_roots.keys():
-        options[ens] = {}
-        for source, table, relpath in zip(
-            [GraphSource.CONTAINMENT_MASS, GraphSource.CONTAINMENT_ACTUAL_VOLUME],
-            [mass_table, actual_volume_table],
-            [mass_relpath, volume_relpath],
-        ):
-            real = table[ens].realizations()[0]
-            options[ens][source] = read_menu_options(table[ens], real, relpath)
-        options[ens][GraphSource.UNSMRY] = {
-            "zones": [],
-            "regions": [],
-            "phases": ["total", "gas", "aqueous"],
+        options[ens] = {
+            GraphSource.CONTAINMENT_MASS: mass_table[ens].menu_options,
+            GraphSource.CONTAINMENT_ACTUAL_VOLUME: actual_volume_table[ens].menu_options,
+            GraphSource.UNSMRY: unsmry_providers[ens].menu_options,
```

Comment:
If unsmry_relpath is not provided this yields an error since unsmry_providers[ens] is None.